recorder.go

```diff
@@ -142,5 +157,11 @@ func (r *SpanRecorder) SendSpans() error {
 		r.spans = nil
 	}
 
+	if resp.StatusCode < 400 {
```

Comment:
We should retry on 5xx errors, and give up the payload on 4xx error

Reply:
Ok, I added a retry algorithm